mkdir -p /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/cnvkit
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/cnvkit batch /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -n /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -f /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa --annotate /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/cnvkit/refFlat.hg19.txt -d /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/cnvkit --scatter --diagram -m amplicon -t /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.total.bed
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/cnvkit.py call -v /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.merged.filt1.vcf.gz -o /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/cnvkit/tumor.sorted.markdup.BQSR.call.cns /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/cnvkit/tumor.sorted.markdup.BQSR.cns
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/cnvkit export vcf /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/cnvkit/tumor.sorted.markdup.BQSR.call.cns -o /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/cnvkit/tumor.somatic.vcf && cp /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/cnvkit/tumor.sorted.markdup.BQSR.call.cns /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/cnv_sv/tumor.cnvkit.somatic.cns

## [20191212 update] cnvkit.py genemetrics
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/cnvkit genemetrics /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/cnvkit/tumor.sorted.markdup.BQSR.cnr -s /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/cnvkit/tumor.sorted.markdup.BQSR.call.cns > /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/cnvkit/tumor.sorted.markdup.BQSR.CNV
/media/gsadmin/vd3/mengxiangfu/software/miniconda3/envs/workenv/bin/python /media/gsadmin/vd3/mengxiangfu/project/1.cancer/CNVdetect_1206/scripts/easyCNVtrans.py /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/cnvkit/tumor.sorted.markdup.BQSR.CNV && cp /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/cnvkit/tumor.sorted.markdup.BQSR.gene.CNV /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/cnv_sv/tumor.cnvkit.somatic.gene.CNV

## [20191225 update] GATK cnv pipeline
/media/gsadmin/vd3/mengxiangfu/project/1.cancer/CNVdetect_1206/scripts/cnv_GATK_ForPipeline.sh /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir

## [2020108 update] GeneFuse
mkdir /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/genefuse
docker run --rm -v /media/gsadmin:/media/gsadmin -w /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir kr1stoff/genefuse:v0.1.1 bash -c "genefuse -r /media/gsadmin/vd3/mengxiangfu/database/reference/hg19/hg19.fa -f /media/gsadmin/vd3/mengxiangfu/project/1.cancer/Fusiondetect_1231/config/druggable.hg19.csv -1 2.result/tumor/clean_data/tumor_R1.fastq.gz -2 2.result/tumor/clean_data/tumor_R2.fastq.gz -h 1.process/tumor/cnvsv/genefuse/tumor.html -j 1.process/tumor/cnvsv/genefuse/tumor.json > 1.process/tumor/cnvsv/genefuse/tumor.result"

mkdir -p /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/lumpy /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/normal/cnvsv/lumpy
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/sambamba view -t 16 -f bam -F "not (proper_pair or unmapped or mate_is_unmapped or secondary_alignment or duplicate)" -o /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/normal/cnvsv/lumpy/normal.discordants.bam /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam && /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/sambamba view -h /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam | /media/gsadmin/vd1/heguangliang/bcbio/anaconda/bin/python /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/extractSplitReads_BwaMem -i stdin | /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/sambamba view -S -t 16 -f bam -o /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/normal/cnvsv/lumpy/normal.splitters.bam /dev/stdin
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/sambamba view -t 16 -f bam -F "not (proper_pair or unmapped or mate_is_unmapped or secondary_alignment or duplicate)" -o /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/lumpy/tumor.discordants.bam /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam && /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/sambamba view -h /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam | /media/gsadmin/vd1/heguangliang/bcbio/anaconda/bin/python /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/extractSplitReads_BwaMem -i stdin | /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/sambamba view -S -t 16 -f bam -o /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/lumpy/tumor.splitters.bam /dev/stdin
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/lumpyexpress -B /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam,/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -S /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/lumpy/tumor.splitters.bam,/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/normal/cnvsv/lumpy/normal.splitters.bam -D /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/lumpy/tumor.discordants.bam,/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/normal/cnvsv/lumpy/normal.discordants.bam -o /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/lumpy/tumor.somatic.vcf
mkdir -p /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/gridss
bash /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gridss_somatic_target.sh /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.total.bed /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/gridss/gridss
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/lumpy_annotation.py -v /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/lumpy/tumor.somatic.vcf -i /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.total.bed -o /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/lumpy/lumpy
mkdir -p /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/merge
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/merge_gridss_lumpy.pl /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/gridss/gridss.final.annotation.txt /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/lumpy/lumpy.final.annotation.txt >/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/merge/merge.final.annotation.txt
perl /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/merge_gridss_lumpy.pl /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/gridss/gridss.final.annotation.filted.txt /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/lumpy/lumpy.final.annotation.filted.txt >/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/merge/merge.final.annotation.filted.txt
cp /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/merge/merge.final.annotation.txt /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/cnvsv/merge/merge.final.annotation.filted.txt /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/cnv_sv/
echo cnvsv tumor finished >>/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/finishedSampleAnalysis.log
