mkdir -p /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/vardict /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2 /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/strelka
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/VarDict_java -G /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa --nosv --deldupvar -Q 10 -F 0x700 -f 0.001 -N tumor -b "/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam|/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam" -c 1 -S 2 -E 3 /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.total.bed -th 24 |/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/testsomatic.java.R | /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/var2vcf_paired.java.pl -N "tumor|normal" -f 0.001 > /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/vardict/tumor.raw.vcf &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/configureStrelkaSomaticWorkflow.py --targeted --callRegions /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.total.bed.gz --runDir /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/strelka --normalBam /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam --tumorBam /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam --referenceFasta /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa
/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/strelka/runWorkflow.py  -m local -j 8 && wait

/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr1.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr1.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr2.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr2.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr3.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr3.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal && wait
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr4.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr4.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr5.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr5.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr6.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr6.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal && wait
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr7.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr7.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr8.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr8.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr9.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr9.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal && wait
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr10.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr10.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr11.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr11.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr12.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr12.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal && wait
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr13.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr13.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr14.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr14.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr15.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr15.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal && wait
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr16.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr16.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr17.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr17.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr18.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr18.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal && wait
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr19.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr19.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr20.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr20.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr21.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr21.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal && wait
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chr22.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr22.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal &
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk Mutect2  -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.chrX.bed -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chrX.vcf.gz -tumor tumor -pon /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/somatic-hg38_1000g_pon.hg19.sorted.vcf.gz --germline-resource /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/af-only-gnomad.hg19.vcf.gz --af-of-alleles-not-in-resource 0.0000025 --disable-read-filter MateOnSameContigOrNoMappedMateReadFilter --pairHMM AVX_LOGLESS_CACHING_OMP --native-pair-hmm-threads 8 -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam -normal normal && wait
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk MergeVcfs -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr1.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr2.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr3.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr4.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr5.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr6.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr7.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr8.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr9.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr10.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr11.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr12.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr13.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr14.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr15.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr16.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr17.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr18.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr19.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr20.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr21.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chr22.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.chrX.vcf.gz  -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.vcf.gz

/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk GetPileupSummaries -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -V /media/gsadmin/vd1/heguangliang/database/gatk_bundle/hg19/small_exac_common_3.hg19.vcf.gz -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor.getpileupsummaries.table -L /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/splitBed/custom.total.bed
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk CalculateContamination -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor.getpileupsummaries.table -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor.contamination.table --tumor-segmentation /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor.segments.table
/media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk FilterMutectCalls -V /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.vcf.gz --contamination-table /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor.contamination.table --tumor-segmentation /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor.segments.table -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.merged.filt1.vcf.gz
# /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk CollectSequencingArtifactMetrics -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam -R /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/reference/hg19/hg19.fa -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor.artifact -VALIDATION_STRINGENCY LENIENT --FILE_EXTENSION ".txt"
# /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk FilterByOrientationBias -AM "G/T" -AM "C/T" -V /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.merged.filt1.vcf.gz -P /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor.artifact.pre_adapter_detail_metrics.txt -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.merged.filt2.vcf.gz
# /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/gatk FilterAlignmentArtifacts  -V /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.merged.filt2.vcf.gz -I /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam --bwa-mem-index-image /media/gsadmin/vd1/heguangliang/database/hg19/hg19.fa.img  -O /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.merged.filt3.vcf.gz

/media/gsadmin/vd1/heguangliang/bcbio2/anaconda/bin/python /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/merge3vcf.py -m /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/mutect2/tumor._somatic_m2.merged.filt1.vcf.gz -v /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/vardict/tumor.raw.vcf -i /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/strelka/results/variants/somatic.indels.vcf.gz -s /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/tumor/snpindel/strelka/results/variants/somatic.snvs.vcf.gz -o /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools.vcf && bgzip /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools.vcf && tabix -p vcf /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools.vcf.gz
perl /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/convert2annovar.pl -format vcf4old /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools.vcf.gz --outfile /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools.avinput --allelicfrac --includeinfo
perl /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/table_annovar.pl /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools.avinput /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/annovar/hg19 -buildver hg19 -out /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools --remove -nastring -- --polish --thread 14 --csvout  --otherinfo -protocol refGeneWithVer,cytoBand,1000g_ALL.sites.2015_08,avsnp150,clinvar_20180729,cosmic85_coding,cosmic85_noncoding,dbnsfp33a,exac03,gnomad_exome,gnomad_genome,intervar,hgmd,esp6500siv2_all -operation g,r,f,f,f,f,f,f,f,f,f,f,f,f
#perl /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/table_annovar.pl /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools.avinput /media/gsadmin/vd1/heguangliang/pipeline/WGS/database/annovar/hg19 -buildver hg19 -out /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools --remove -nastring -- --polish --thread 14 --csvout  --otherinfo -protocol refGeneWithVer,cytoBand,1000g_ALL.sites.2015_08,avsnp150,clinvar_20180729,cosmic85_coding,cosmic85_noncoding,dbnsfp33a,exac03,gnomad_exome,gnomad_genome,intervar,hgmd,esp6500siv2_all,omim201806 -operation g,r,f,f,f,f,f,f,f,f,f,f,f,f,f
#/media/gsadmin/vd1/heguangliang/bcbio3/anaconda/envs/DNA_pipeline_python2/bin/python /media/gsadmin/vd1/heguangliang/pipeline/WGS/bin/proannovar_tumor.py -i /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools.hg19_multianno.csv -o /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools.hg19_multianno.pro.csv
perl -ne 'if($.==1){print "Tumor_total_depth,Normal_total_depth,Tumor_Alt_frequency,Normal_Alt_frequency,",$_}else{@t=split "\t";my $td=(split ":", $t[-2])[1];my $nd=(split ":", $t[-1])[1];print "$td,$nd,$1,",$_ if m/AF=(.+?);/}' /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools.hg19_multianno.csv >/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools.hg19_multianno.add.csv

## [20200616] tmb / optimize annotation
/media/gsadmin/vd3/mengxiangfu/software/miniconda3/envs/workenv/bin/python /media/gsadmin/vd3/mengxiangfu/project/1.cancer/TMB_optimize_0924/scripts/tmbstatsFor550pipeline_1023.py /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools.hg19_multianno.add.csv
/media/gsadmin/vd3/mengxiangfu/software/miniconda3/envs/workenv/bin/python /media/gsadmin/vd3/mengxiangfu/project/1.cancer/1.550_Panel/annotation/multianno_Custom550Panel.py -i /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_variation/snp_indel/tumor.3tools.hg19_multianno.add.csv

mkdir -p /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/msi /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/report/msi
echo -e "/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/tumor/result_alignment/tumor.sorted.markdup.BQSR.bam\n/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/normal/result_alignment/normal.sorted.markdup.BQSR.bam" >/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/msi/list
bash /media/gsadmin/vd2/data/zl/msings/run_msings.v2.sh /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/msi/list /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/msi && cp /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/1.process/msi/Combined_MSI.txt /media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/2.result/report/msi
echo somatic tumor finished >>/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/finishedSampleAnalysis.log
echo annotate tumor finished >>/media/gsadmin/vd2/data/zl/20181229LYF/Target_hg19_work_dir/finishedSampleAnalysis.log
